TombEngine/Objects/TR4/Trap/SquishyBlock.cpp

```diff
@@ -0,0 +1,204 @@
+#include "framework.h"
+#include "Objects/TR4/Trap/SquishyBlock.h"
+
+#include "Game/animation.h"
+#include "Game/camera.h"
+#include "Game/collision/collide_item.h"
+#include "Game/collision/collide_room.h"
+#include "Game/collision/sphere.h"
+#include "Game/control/control.h"
+#include "Game/effects/effects.h"
+#include "Game/items.h"
+#include "Game/Lara/lara.h"
+#include "Game/misc.h"
+#include "Game/Setup.h"
+#include "Sound/sound.h"
+#include "Specific/level.h"
+
+// NOTES:
+// item.ItemFlags[0]: use dynamic motion.
+// item.ItemFlags[1]: ??
+// item.ItemFlags[4]: heading angle.
+// TODO: Add support for 4 directions.
+
+namespace TEN::Entities::Traps
+{	
+	constexpr auto SQUISHY_BLOCK_LETHAL_FRAME = 33;
+
+	enum SquishyBlockState
+	{
+		SQUISHY_BLOCK_STATE_MOVE = 0,
+		SQUISHY_BLOCK_STATE_COLLIDE_LEFT = 1,
+		SQUISHY_BLOCK_STATE_COLLIDE_RIGHT = 2,
+		SQUISHY_BLOCK_STATE_BAKED_MOTION = 3,
+		SQUISHY_BLOCK_STATE_COLLIDE_FRONT = 4,
+		SQUISHY_BLOCK_STATE_COLLIDE_BACK = 5
+	};
+
+	enum SquishyBlockAnim
+	{
+		SQUISHY_BLOCK_ANIM_MOVE = 0,
+		SQUISHY_BLOCK_ANIM_IMPACT_BACK = 1,
+		SQUISHY_BLOCK_ANIM_IMPACT_FRONT = 2,
+		SQUISHY_BLOCK_ANIM_BAKED_MOTION = 3,
+		SQUISHY_BLOCK_ANIM_COLLIDE_FRONT = 4,
+		SQUISHY_BLOCK_ANIM_COLLIDE_BACK = 5
+	};
+
+	void InitializeSquishyBlock(short itemNumber)
+	{
+		auto& item = g_Level.Items[itemNumber];
+
+		if (!item.TriggerFlags)
+			SetAnimation(item, SQUISHY_BLOCK_ANIM_BAKED_MOTION);
+
+		item.HitPoints = NOT_TARGETABLE;
+		item.ItemFlags[0] = item.TriggerFlags;
+		item.ItemFlags[1] = 0;
+		item.ItemFlags[4] = 0;
+	}
+
+	void ControlSquishyBlock(short itemNumber)
+	{
+		auto& item = g_Level.Items[itemNumber];
+		const auto& object = Objects[item.ObjectNumber];
+
+		short& something0 = item.ItemFlags[0];
+		short& someAngle = item.ItemFlags[1];
+		short& headingAngle = item.ItemFlags[4];
+
+		if (!TriggerActive(&item))
+			return;
+
+		if (!item.TriggerFlags)
+		{
+			if (item.Animation.ActiveState != SQUISHY_BLOCK_STATE_BAKED_MOTION)
+				SetAnimation(item, SQUISHY_BLOCK_ANIM_BAKED_MOTION);
+		}
+		else
+		{
+			if (item.Animation.ActiveState == SQUISHY_BLOCK_ANIM_BAKED_MOTION)
+				SetAnimation(item, SQUISHY_BLOCK_STATE_MOVE);
+
+			something0 = item.TriggerFlags;
```

Comment:
This variable name is totally non-descriptive, please give it a meaningful name.

Reply:
fixed